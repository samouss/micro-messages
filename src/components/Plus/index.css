.Plus:before,
.Plus:after {
  content: '';
  width: 20px;
  height: 2px;
  background: #FFFFFF;
  display: block;
  position: absolute;
  left: 50%;
}

.Plus:before {
  transform: translateX(-50%);
}

.Plus:after {
  transform: translateX(-50%) rotate(90deg);
}
