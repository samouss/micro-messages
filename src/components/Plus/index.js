// @flow

import React from 'react';
import './index.css';

export type Props = {
  customClassName?: string,
};

const Plus = ({ customClassName = '', ...props }: Props) => (
  <div
    {...props}
    styleName="Plus"
    className={customClassName}
  />
);

export default Plus;
