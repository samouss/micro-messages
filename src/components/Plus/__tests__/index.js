// @flow

import React from 'react';
import { shallow } from 'enzyme';
import Plus from '../index';

import type { Props } from '../index';

describe('<Plus />', () => {
  const defaultProps: Props = {};

  it('expect to render the component correctly', () => {
    const props: Props = {
      ...defaultProps,
    };

    const component = shallow(
      <Plus
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to render the component with custom className', () => {
    const props: Props = {
      ...defaultProps,
      customClassName: 'someClass',
    };

    const component = shallow(
      <Plus
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });
});
