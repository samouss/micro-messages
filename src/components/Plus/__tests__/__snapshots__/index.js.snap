// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Plus /> expect to render the component correctly 1`] = `
<div
  className=""
  styleName="Plus"
/>
`;

exports[`<Plus /> expect to render the component with custom className 1`] = `
<div
  className="someClass"
  styleName="Plus"
/>
`;
